tests/lib/nm/dns_test.py

```diff
@@ -124,3 +125,46 @@ def _assert_dns(setting_ip, dns_conf):
     priority = dns_conf.get(nm_dns.DNS_METADATA_PRIORITY,
                             nm_dns.DEFAULT_DNS_PRIORITY)
     assert setting_ip.props.dns_priority == priority
+
+
+def test_find_interfaces_for_dns_with_ipv6_and_ipv4_static_gateways():
+    iface_routes = _get_test_iface_routes()
+    assert (('eth1', 'eth2') ==
+            nm_dns.find_interfaces_for_name_servers(iface_routes))
+
+
+def test_find_interfaces_for_dns_with_ipv6_static_gateway_only():
+    iface_routes = _get_test_iface_routes()
+    del iface_routes['eth1']
+    assert ((None, 'eth2') ==
+            nm_dns.find_interfaces_for_name_servers(iface_routes))
+
+
+def test_find_interfaces_for_dns_with_ipv4_static_gateway_only():
+    iface_routes = _get_test_iface_routes()
+    del iface_routes['eth2']
+    assert (('eth1', None) ==
+            nm_dns.find_interfaces_for_name_servers(iface_routes))
+
```

Comment:
two tests are missing:
- There are no routes at all.
- There is one a non gateway route.